Core/Core/Extensions/URLComponentsExtensions.swift

```diff
@@ -46,6 +46,17 @@ public extension URLComponents {
         return url
     }
 
+    var queryMap: [String: String] {
+        let items = queryItems ?? []
+        return items.reduce([:]) { result, item in
+            var result = result
+            if let value = item.value {
+                result[item.name] = value
+            }
+            return result
+        }
+    }
```

Comment:
I don't think we want this, since a query item name can be repeated many times, and the resulting dictionary will only have the last instance.